### PR TITLE
Markdown Reference Page

### LINKED ISSUE #27: Create docs page Reference/Material for MkDocs
### Purpose
What you would need to know to contribute to our documentation sites

### Page Tree
```yml
nav:
  ...
  - Reference:
    ...
    - Material for MkDocs: reference/material-for-mkdocs.md
```

### Contents
- Relevant files
    - docs/
    - mkdocs.yml
        - Page tree
- Reference demo/syntax
- GitHub pages integration
   - GitHub actions
   - Repository setup

### Relevant Resources
- https://patrick-5546.github.io/notes/reference/mkdocs/
- https://squidfunk.github.io/mkdocs-material/reference/

### LINKED ISSUE #36: Create docs page Reference/Markdown
### Purpose
Markdown syntax for GitHub (README, issues, PR's)

### Page Tree
```yml
nav:
  ...
  - Reference:
    ...
    - Markdown: reference/markdown.md
```

### Resources
- https://www.markdownguide.org/getting-started/
- https://docs.github.com/en/get-started/writing-on-github

### PR DESCRIPTION
**Pull request by DFriend01**
Resolves #36.

## What is Included
I have included the following topics:
- What is Markdown?
- Getting Started
- Where and how we use Markdown on Sailbot

I also linked to the necessary resources for tutorials on Markdown and references for the different flavors of Markdown that we use.

### Markdown Linting and Checks
I chose **not** to include a section on the `.json` files in the repository related to Markdown. I feel like these topics are more appropriate for either the section on MkDocs or VS Code. Let me know what your thoughts are.

### VS Code Extensions
I briefly considered talking about some helpful VS Code extensions for Markdown, but I was not sure if it was particularly relevant. We also don't use extensions for Markdown in the Sailbot Workspace if I recall correctly. What do you think? I was thinking of the following extensions:
- [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one)
- [Markdown Preview GitHub Styling](https://marketplace.visualstudio.com/items?itemName=bierner.markdown-preview-github-styles)

### CODE REVIEW

File: docs/reference/markdown/markdown.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Markdown
+
+Markdown is a lightweight markup language that you can use to add formatting elements to plaintext text documents[^1].
```
Comment (patrick-5546) [inline docs/reference/markdown/markdown.md]: Footnote after punctuation

File: docs/reference/markdown/markdown.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+Almost anywhere in GitHub where you can write text, you can format it with Markdown. GitHub also allows you to preview
+your Markdown before you submit any comments.
+
+=== "Before Rendering"
+
+    ![image](../../assets/images/markdown/issue_md_text.png)
+
+=== "After Rendering"
+
+    ![image](../../assets/images/markdown/issue_md_rendered.png)
```
Comment (patrick-5546) [inline docs/reference/markdown/markdown.md]: Fancy, I like it!

File: docs/reference/markdown/markdown.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+### Material for MkDocs
+
+We use Markdown with Material for MkDocs to create this website! Everything that you see here has been rendered with
+Markdown. This allows us to create this site without any frontend experience (although, we could spice up this site
+even further with HTML/CSS)!
+
+Material for MkDocs supports most of the common Markdown syntax. However, it also adds on to its flavour of Markdown
+by adding many of its own customized features. Feel free to look at the [source code](https://github.com/UBCSailbot/docs)
+for this site for how our pages were generated using MkDocs-flavoured Markdown.
```
Comment (patrick-5546) [inline docs/reference/markdown/markdown.md]: Consider linking to the `docs/` folder, where our site's pages are contained

File: docs/reference/markdown/markdown.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+!!! note "MkDocs-Flavoured Markdown"
```
Comment (patrick-5546) [inline docs/reference/markdown/markdown.md]: Material for MkDocs, what we use, is a subset of MkDocs. It makes it much quicker and easier to create beautiful documentation.

File: docs/reference/markdown/markdown.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+    Many of the features of MkDocs can be utilized through its flavour of Markdown. Refer the the
```
Comment (patrick-5546) [inline docs/reference/markdown/markdown.md]: Consider changing first sentence to something akin to:

Material for MkDocs' flavour of Markdown extends upon vanilla Markdown, adding features that can be used to take your technical documentation to the next level.

**Review by patrick-5546 — COMMENTED** [review]
Sorry for the spam, I forgot to add comments to a review.

It looks great! I'm thinking we can close out #27 (link it to this PR), I don't think we need a separate page to go over Material for MkDocs considering its official documentation is great and it's intuitive to learn (especially since we have pages in our repo to refer to).

Addressing your PR description comments:
- I think linting, including markdown, should go in the vscode page
- I don't like all in one, but I haven't seen the GitHub preview before and it looks amazing. I added it to Eric's PR's in docs and sailbot_workspace that update our vscode extensions

File: docs/reference/markdown/markdown.md (hunk 2 above)
Comment (DFriend01) [reply docs/reference/markdown/markdown.md]: $T_HAN^KS$

File: docs/reference/markdown/markdown.md (hunk 3 above)
Comment (DFriend01) [reply docs/reference/markdown/markdown.md]: Done

**Comment by DFriend01** [comment]
We already link to the documentation for Material for MkDocs. It is probably fair to say that this PR also resolves #27.

**Review by patrick-5546 — COMMENTED** [review]
I changed my mind, I think we should have a section on linting to match our Python and C++ documentation. It also might be helpful to go over different methods of rendering Markdown locally. In the VS Code page I will recommend specific extensions for each language.

Notes:
- I forced pushed to remove the infrastructure changes, putting them in #71 instead
- Both GitHub and Material docs fully support vanilla Markdown, they only add/extend functionally

**Comment by DFriend01** [comment]
@patrick-5546 I added the following additional sections to the Markdown page:
- Rendering Markdown Locally
- Markdown Linting

Also, the CI seems to be failing on a working link. I am thinking of adding that link to [.markdown-link-check.json](https://github.com/UBCSailbot/docs/blob/main/.markdown-link-check.json) under the ignore list.

UPDATE: I added an ignore pattern to ignore the GitHub docs links.

File: docs/reference/markdown.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    (like this note) and content tabs. Refer to the
+    [official Material for MkDocs reference page](https://squidfunk.github.io/mkdocs-material/reference/) for more
+    information on the available features.
+
+## Rendering Markdown Locally
+
+You have a few choices to render Markdown on your computer.
+Be advised that if you are using an extended version of Markdown, you will
+need to consult the documentation from the service provider to render their flavour of Markdown properly. The following
+resources are good for rendering vanilla Markdown:
```
Comment (patrick-5546) [inline docs/reference/markdown.md]: I'm thinking we can this section to just "Rendering Markdown"
- Vanilla markdown options are good
- For GitHub, recommend drafting an issue or something and the GitHub preview VS Code extension
- For mkdocs, recommend running the site locally  with `mkdocs serve`

**Review by patrick-5546 — APPROVED** [review]
Great work on this PR, feel free to merge

**Pull request merged**